src/functional/ffront/type_info.py

```diff
@@ -65,154 +252,51 @@ def function_signature_incompatibilities(
         yield f"Got unexpected keyword argument(s) `{'`, `'.join(kw_b_m_a)}`."
 
     for kwarg in set(func_type.kwargs.keys()) & set(kwargs.keys()):
-        if func_type.kwargs[kwarg] != kwargs[kwarg]:
+        if (a_kwarg := func_type.kwargs[kwarg]) != (
+            b_kwarg := kwargs[kwarg]
+        ) and not can_concretize(a_kwarg, to_type=b_kwarg):
             yield f"Expected keyword argument {kwarg} to be of type {func_type.kwargs[kwarg]}, but got {kwargs[kwarg]}."
 
 
-@dataclass
-class TypeInfo:
-    """
-    Wrapper around foast types for type deduction and compatibility checks.
-
-    Examples:
-    ---------
-    >>> type_a = ScalarType(kind=ScalarKind.FLOAT64)
-    >>> typeinfo_a = TypeInfo(type_a)
-    >>> typeinfo_a.is_complete
-    True
-    >>> typeinfo_a.is_arithmetic_compatible
-    True
-    >>> typeinfo_a.is_logics_compatible
-    False
-    >>> typeinfo_b = TypeInfo(None)
-    >>> typeinfo_b.is_any_type
-    True
-    >>> typeinfo_b.is_arithmetic_compatible
-    False
-    >>> typeinfo_b.can_be_refined_to(typeinfo_a)
-    True
-
+def can_call(
```

Comment:
I am generally not a fan of the term `can` when naming functions and it is in my experience more common to use `is` + an adjectival suffix (-able, -like). I see the case for making this look sentence like, but I do not endorse this style as it inevitably breaks and thus can not be done consistently anyway.

Reply:
I honestly do not care, `is_<adjective>` style is more consistent with what we have elsewhere. That's probably more important than the longer names it leads to.